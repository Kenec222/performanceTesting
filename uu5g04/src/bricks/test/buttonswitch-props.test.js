/**
 * Copyright (C) 2021 Unicorn a.s.
 *
 * This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public
 * License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License at
 * <https://gnu.org/licenses/> for more details.
 *
 * You may obtain additional information at <https://unicorn.com> or contact Unicorn a.s. at address: V Kapslovne 2767/2,
 * Praha 3, Czech Republic or at the email: info@unicorn.com.
 */

import UU5 from "uu5g04";
import "uu5g04-bricks";
import { mount, shallow, omitConsoleLogs } from "uu5g05-test";

beforeEach(() => {
  omitConsoleLogs('"size" is deprecated');
  omitConsoleLogs('"bgStyle" is deprecated');
});

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.NestingLevelMixin",
    "UU5.Common.ContentMixin",
    "UU5.Common.PureRenderMixin",
  ],
  props: {
    size: {
      values: ["s", "m", "l", "xl"],
    },
    switchedOn: {
      values: [true, false],
    },
    bgStyle: {
      values: ["filled", "outline", "transparent"],
    },
    props: {
      values: [{ size: "s" }],
    },
    onProps: {
      values: [{ colorSchema: "success", pressed: true }],
    },
    offProps: {
      values: [{ colorSchema: "danger", pressed: false }],
    },
  },
  requiredProps: {},
  opt: {},
};

describe(`UU5.Bricks.ButtonSwitch props testing`, () => {
  UU5.Test.Tools.testProperties(UU5.Bricks.ButtonSwitch, CONFIG);

  const This = {};

  it(`UU5.Bricks.ButtonSwitch with combination of props, onProps, onProps`, () => {
    const wrapper = shallow(
      <UU5.Bricks.ButtonSwitch
        id={"switchID"}
        ref_={(item) => (This._buttonSwitch = item)}
        props={{
          onClick: () => This._buttonSwitch.toggle(),
          content: "props content",
        }}
        onProps={{
          content: "onProps content",
        }}
        offProps={{
          content: "offProps content",
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.simulate("click");
    expect(wrapper).toMatchSnapshot();
  });

  //If I click to buttonSwitch the content turned off button it turns to turned on
  it(`UU5.Bricks.ButtonSwitch with onClick`, () => {
    const wrapper = shallow(
      <UU5.Bricks.ButtonSwitch
        id={"switchID"}
        ref_={(item) => (This._buttonSwitch = item)}
        props={{
          onClick: () => This._buttonSwitch.toggle(),
        }}
        onProps={{
          colorSchema: "default",
          content: "turned on",
        }}
        offProps={{
          content: "turned off",
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().state.switchedOn).toBeFalsy();
    wrapper.simulate("click");
    expect(wrapper.instance().state.switchedOn).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
