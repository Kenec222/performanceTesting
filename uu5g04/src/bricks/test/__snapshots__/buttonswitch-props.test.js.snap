// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.ButtonSwitch props testing UU5.Bricks.ButtonSwitch with combination of props, onProps, onProps 1`] = `
<Button
  className="uu5-bricks-button-switch"
  content="offProps content"
  id="switchID-inner"
  mainAttrs={Object {}}
  onClick={[Function]}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing UU5.Bricks.ButtonSwitch with combination of props, onProps, onProps 2`] = `
<Button
  className="uu5-bricks-button-switch"
  content="onProps content"
  id="switchID-inner"
  mainAttrs={Object {}}
  onClick={[Function]}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing UU5.Bricks.ButtonSwitch with onClick 1`] = `
<Button
  className="uu5-bricks-button-switch"
  content="turned off"
  id="switchID-inner"
  mainAttrs={Object {}}
  onClick={[Function]}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing UU5.Bricks.ButtonSwitch with onClick 2`] = `
<Button
  className="uu5-bricks-button-switch"
  colorSchema="default"
  content="turned on"
  id="switchID-inner"
  mainAttrs={Object {}}
  onClick={[Function]}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing bgStyle="filled" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing bgStyle="outline" 1`] = `
<Button
  bgStyle="outline"
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing bgStyle="transparent" 1`] = `
<Button
  bgStyle="transparent"
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing checkGrammar=false 1`] = `
<Button
  checkGrammar={false}
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing checkGrammar=true 1`] = `
<Button
  checkGrammar={true}
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing checkHighlight=false 1`] = `
<Button
  checkHighlight={false}
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing checkHighlight=true 1`] = `
<Button
  checkHighlight={true}
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing checkSpaces=false 1`] = `
<Button
  checkSpaces={false}
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing checkSpaces=true 1`] = `
<Button
  checkSpaces={true}
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing className="my-classname" 1`] = `
<Button
  className="uu5-bricks-button-switch my-classname"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing className="test-classname" 1`] = `
<Button
  className="uu5-bricks-button-switch test-classname"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing content="This is Content from ContentMixin" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  content="This is Content from ContentMixin"
  id="uuID-inner"
  mainAttrs={Object {}}
>
  <UU5.Common.TextCorrectorContextConsumer
    key="0"
    text="This is Content from ContentMixin"
  />
</Button>
`;

exports[`UU5.Bricks.ButtonSwitch props testing content=12345 1`] = `
<Button
  className="uu5-bricks-button-switch"
  content={12345}
  id="uuID-inner"
  mainAttrs={Object {}}
>
  12345
</Button>
`;

exports[`UU5.Bricks.ButtonSwitch props testing controlled=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing controlled=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing default props 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing disabled=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing disabled=false 3`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing disabled=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  disabled={true}
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing disabled=true 3`] = `
<Button
  className="uu5-bricks-button-switch"
  disabled={true}
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing dynamic=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing dynamic=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  dynamic={true}
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing elementAttrs={"data":{"test":"data-test-value"}} 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing fullTextSearchPriority=0 1`] = `
<Button
  className="uu5-bricks-button-switch uu-fulltextsearch-0"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing fullTextSearchPriority=0 2`] = `
<Button
  className="uu5-bricks-button-switch uu-fulltextsearch-0"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing hidden=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing hidden=false 3`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing hidden=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  hidden={true}
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing hidden=true 3`] = `
<Button
  className="uu5-bricks-button-switch"
  hidden={true}
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing id="myId" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="myId-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing id="test-id" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="test-id-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing ignoreHTML=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing ignoreHTML=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing mainAttrs=undefined 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing name="myName" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  name="myName-inner"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="bigBox" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBox"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="bigBoxCollection" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="box" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="box"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="box" 3`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="box"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="boxCollection" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="boxCollection"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="inline" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="inline"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="inline" 3`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="inline"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="page" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="page"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="smallBox" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="smallBox"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="smallBoxCollection" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="smallBoxCollection"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing nestingLevel="spa" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="spa"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing noPrint=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing noPrint=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing offProps={"colorSchema":"danger","pressed":false} 1`] = `
<Button
  className="uu5-bricks-button-switch"
  colorSchema="danger"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing onProps={"colorSchema":"success","pressed":true} 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing props={"size":"s"} 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  size="s"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing pureRender=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing pureRender=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  pureRender={true}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing selected=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing selected=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  selected={true}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing size="l" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  size="l"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing size="m" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing size="s" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  size="s"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing size="xl" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  size="xl"
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing style="padding:8px; borderRadius: 2px" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing style={"backgroundColor":"red","color":"blue"} 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing style={"margin":8} 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  style={
    Object {
      "margin": 8,
    }
  }
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing switchedOn=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing switchedOn=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing textCorrector=false 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  textCorrector={false}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing textCorrector=true 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={Object {}}
  textCorrector={true}
/>
`;

exports[`UU5.Bricks.ButtonSwitch props testing tooltip="My Tooltip" 1`] = `
<Button
  className="uu5-bricks-button-switch"
  id="uuID-inner"
  mainAttrs={
    Object {
      "title": "My Tooltip",
    }
  }
  tooltip="My Tooltip"
/>
`;
